compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp

```diff
@@ -27,6 +28,103 @@ namespace Flow {
 
 namespace {
 
+LogicalResult tryRaiseSlicingOp(linalg::LinalgOp linalgOp,
+                                RewriterBase &rewriter) {
+  if (!linalgOp.hasTensorSemantics())
+    return failure();
+  // Check if there is only one result.
+  if (!llvm::hasSingleElement(linalgOp.getOperation()->getResults()))
+    return failure();
+  // Check if operation is elementwise. For now, we do not handle elementwise
+  // with permutations.
+  if (!isElementwise(linalgOp))
+    return failure();
+  if (!linalgOp.getIndexingMapsArray().back().isIdentity())
+    return failure();
+
+  // The outs should be a tensor.
+  auto initTensor =
+      dyn_cast<TensorType>(linalgOp.getDpsInitOperand(0)->get().getType());
+  if (!initTensor)
+    return failure();
+
+  linalg::YieldOp yieldOp =
+      cast<linalg::YieldOp>(linalgOp.getBlock()->getTerminator());
+
+  if (!yieldOp.getOperand(0).getDefiningOp())
+    return failure();
+
+  auto extractOp =
+      dyn_cast<tensor::ExtractOp>(yieldOp.getOperand(0).getDefiningOp());
+  if (!extractOp)
+    return failure();
+
+  // Get the tensor being extracted from.
+  TypedValue<RankedTensorType> source = extractOp.getTensor();
+
+  // Slicing does not change rank.
+  if (initTensor.getRank() == source.getType().getRank())
```

Comment:
The condition is flipped here; it should be checking whether the ranks differ. Also you can add a TODO for supporting rank-reducing slicing.